Feat/metamask provider

Sednaoui opened the pull request.

Sednaoui committed:
add packages:
- @metamask/post-message-stream
-@metamask/providers
-loglevel

Sednaoui committed:
add ether.js package

Sednaoui committed:
add redux packages:
* @reduxjs/toolkit
* react-redux
* redux-saga-routines
* redux-saga
* @types/react-redux
* @types/redux-saga-routines

Sednaoui committed:
create store for wallet instance and connect to App
* wallet reducer
* wallet actions
* wallet saga
* wallet types
* pass provider to App

QuocAnhVu committed:
Added hot-reloading injected script.

Committed alongside modifications to react-browser-extension-scripts package in a seperate repository: https://github.com/QuocAnhVu/create-react-extension/tree/feat/injectedScript.

Sednaoui commented on crosscash-extension/package.json:
```diff
@@ -14,9 +15,8 @@
-    "react-browser-extension-scripts": "4.0.10",
     "react-dom": "^17.0.2",
-    "typescript": "^4.5.3",
+    "typescript": "^4.5.4",
```
here

Sednaoui committed:
add redux logger

Sednaoui committed:
create wallet modal

Sednaoui committed:
add unit tests for wallet model

Sednaoui committed:
add redux-persist:
* local storage and persistance after closing browser

Sednaoui committed:
Merge pull request #5 from QuocAnhVu/feat/create-wallet-utilities

Feat/create wallet utilities

QuocAnhVu committed:
Added provider to injected script and demonstrated communication between inpage provider and contentScript

QuocAnhVu committed:
Injected script now runs before dApp scripts so that window.ethereum is ready on initialization

QuocAnhVu commented:
You can open the test dApp and type `window.stream.write('hey');` in the console.

The button will also switch from "Click here to install metamask" to "Connect", signaling that the dApp detects the provider.

Sednaoui commented:
> You can open the test dApp and type `window.stream.write('hey');` in the console.
> 
> The button will also switch from "Click here to install metamask" to "Connect", signaling that the dApp detects the provider.

I got 'hey' back 🥳

for some reason I never got the button to install metamask even without any extension installed on my browser. It was always connect.

QuocAnhVu committed:
Renamed 'injected' to 'inpage' to be more precise and less scary

Companion commit: https://github.com/QuocAnhVu/create-react-extension/commit/870a6d6b136bb2471a1d42418179968b81c222b8

QuocAnhVu committed:
Merge branch 'feat/metamask-provider'

QuocAnhVu commented:
ready to merge
